model/interactive/src/main/proto/beam_interactive_api.proto

```diff
@@ -0,0 +1,178 @@
+
+  // Responds with debugging and other cache-specific metadata.
+  rpc Status (StatusRequest) returns (StatusResponse) {}
+}
+
+// The state of the InteractiveService. The default state is STOPPED.
+enum State {
+  // The InteractiveService is not replaying. Goes to RUNNING with a
+  // StartRequest.
+  STOPPED = 0;
+
+  // The InteractiveService is replaying events. Goes to PAUSED with a
+  // PauseRequest. Goes to STOPPED with a StopRequest.
+  RUNNING = 1;
+
+  // The InteractiveService is paused from replaying events. Goes to RUNNING
+  // with either a StartRequest or a StepRequest. Goes to STOPPED with a
+  // StopRequest.
+  PAUSED = 2;
+
+  // The InteractiveService is stepping through a single event. Will move to
+  // PAUSED after quiescence.
+  STEPPING = 3;
+
+
+  // The InteractiveService is advancing until a specified duration is reached.
+  // Will move to PAUSED after the stream advances sufficiently.
+  ADVANCING = 4;
+}
+
+message StartRequest {
+  // (Optional) How quickly the stream will be played back, e.g. if
+  // playback_speed == 2, then the stream will replay events twice as fast as
+  // they were recorded. If unspecified, this will default to 1.
+  double playback_speed = 1;
+}
+message StartResponse { }
+
+message AdvanceRequest {
+  // (Required) Will advance the stream by replaying events as quickly as
+  // possible until the stream timestamp has advanced by the specified amount.
+  google.protobuf.Duration advance_by = 1;
+}
+message AdvanceResponse {}
+
+message StopRequest { }
+message StopResponse { }
+
+message PauseRequest { }
+message PauseResponse {
+  // The current timestamp of the replay stream.
+  google.protobuf.Timestamp stream_time = 1;
+
+  // The minimum watermark across all of the faked replayable unbounded sources.
+  google.protobuf.Timestamp watermark = 2;
+}
+
+message StatusRequest { }
+message StatusResponse {
+  // The current timestamp of the replay stream. Is MIN_TIMESTAMP when state
+  // is STOPPED.
+  google.protobuf.Timestamp stream_time = 1;
+
+  // The minimum watermark across all of the faked replayable unbounded sources.
+  // Is MIN_TIMESTAMP when state is STOPPED.
+  google.protobuf.Timestamp watermark = 2;
+
+  // The latest timestamp of the recording stream. Is MIN_TIMESTAMP if there is
+  // no recording.
+  google.protobuf.Timestamp recording_time = 3;
+
+  // The set playback_speed from the StartRequest. Playback speed is set by
+  // StartRequest, or if the stream_time is the current time and the recording
+  // is still happening, the playback speed is 1, else 0.
+  double playback_speed = 4;
+
+  // The state of the InteractiveService. The default state is STOPPED.
+  State state = 5;
+}
+
+message StepRequest {
+  // (Optional) The count of events to stream. If unspecified, the default is
+  // 1, i.e. only read a single element from any source and send it through
+  // the pipeline.
+  int64 count = 1;
+}
+message StepResponse { }
```

Comment:
Should this also return stream_time and watermark?

Reply:
sgtm